src/main/java/com/linkedin/kmf/services/CommitLatencyMetrics.java

```diff
@@ -41,14 +42,47 @@
     _commitOffsetLatency.add(new MetricName("commit-offset-latency-ms-avg", METRIC_GROUP_NAME, "The average latency in ms of committing offset", tags), new Avg());
     _commitOffsetLatency.add(new MetricName("commit-offset-latency-ms-max", METRIC_GROUP_NAME, "The maximum latency in ms of committing offset", tags), new Max());
 
+    if (latencyPercentileGranularityMs == 0) {
+      throw new IllegalArgumentException();
+    }
     int bucketNum = latencyPercentileMaxMs / latencyPercentileGranularityMs + 2;
     int sizeInBytes = bucketNum * 4;
     _commitOffsetLatency.add(new Percentiles(sizeInBytes, latencyPercentileMaxMs, Percentiles.BucketSizing.CONSTANT,
         new Percentile(new MetricName("commit-offset-latency-ms-99th", METRIC_GROUP_NAME, "The 99th percentile latency of committing offset", tags), 99.0),
         new Percentile(new MetricName("commit-offset-latency-ms-999th", METRIC_GROUP_NAME, "The 99.9th percentile latency of committing offset", tags), 99.9),
         new Percentile(new MetricName("commit-offset-latency-ms-9999th", METRIC_GROUP_NAME, "The 99.99th percentile latency of committing offset", tags), 99.99)));
-
     LOG.info("{} was constructed successfully.", this.getClass().getSimpleName());
+  }
+
+  /**
+   * start the recording of consumer offset commit
+   */
+  public void recordCommitStart() {
```

Comment:
I would suggest you consider adding some defensive mechanisms to make sure the `recordCommitStart ()` is always called before `recordCommitComplete()` is called. I prefer this kind of defensive programming mindset especially when methods/APIs are expected to be called/invoked in a particular order. 

This defensive mindset assumes that users of the class (or the caller) might not behave as expected. In this case, that means not calling `recordCommitStart ()`, and `recordCommitComplete()` in an interleaving way. In general, it helps with debugging if something goes wrong.

Reply:
Let's implement a defensive mechanisms ensuring `recordCommitStart()` is called before `recordCommitComplete()` is trigggered.